packages/integration-sdk-runtime/src/execution/__tests__/executeIntegration.test.ts

```diff
@@ -913,13 +913,24 @@ describe('executeIntegrationInstance', () => {
       },
     });
 
-    await expect(
-      executeIntegrationInstance(
-        config.logger,
-        config.instance,
-        config.invocationConfig,
-      ),
-    ).rejects.toThrow(/Duplicate _key detected \(_key=key_a\)/);
+    const response = await executeIntegrationInstance(
+      config.logger,
+      config.instance,
+      config.invocationConfig,
+    );
+    expect(response).toMatchObject({
+      integrationStepResults: [
+        {
+          encounteredTypes: ['duplicate_entity'],
+          status: 'failure',
+        },
+      ],
+      metadata: {
+        partialDatasets: {
+          types: [],
+        }
```

Comment:
In this case, does the one successful `duplicate_entity` get sent to the persister? If so, `duplicate_entity` should be specified as `partial`.